### PR TITLE
Faceted filters testcases

### PR DESCRIPTION
**Pull request by maryvictol**
This PR adds testcases for "Advanced search and faceted filters" (#1829, #1836, #1837, #1838)

### CODE REVIEW

File: docs/testcases/UI/Advanced_Search/1837_3.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+**Actions**:
+
+1. Click on **Search** icon at the left menu bar.
+2. Navigate to `Dict1` filter.
+3. Select `Value1` control.
+4. Hover over the found `folder1` in an appeared list
+5. `Ctrl +` Click on the found `folder1` name
+6. Click **Table** button to switch page to table view
+7. Click **Clear filters** button
+8. Enter `folder1` name into the searh string
```
Comment (NShaforostov) [inline docs/testcases/UI/Advanced_Search/1837_3.md]: > ... the searh ...

Fix to "the search"

File: docs/testcases/UI/Advanced_Search/1837_3.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+8. Create pipeline (`pipeline1`)
+9. Create storage (`storage1`)
+10. Open `storage1` and create file (`file1`)
+
+**Actions**:
+
+1. Click on **Search** icon at the left menu bar.
+2. Navigate to `Dict1` filter.
+3. Select `Value1` control.
+4. Hover over the found `folder1` in an appeared list
```
Comment (NShaforostov) [inline docs/testcases/UI/Advanced_Search/1837_3.md]: Change the action to the actual current one (_click the 'info' icon_)

File: docs/testcases/UI/Advanced_Search/1837_3.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+After step 4: preview panel appears that contains:
+- Folder name that is equal to the name `folder1`
+- Attribute `Dict1` with value `Value1`
+- `pipeline1` and `storage1` created at Preparations steps
+
+After step 5: the `folder1` page opens in the new browser tab
+
+After step 6: `Value1` is shown in `Dict1` column for `folder1`
+
+After step 3:
```
Comment (NShaforostov) [inline docs/testcases/UI/Advanced_Search/1837_3.md]: Fix number of the step

File: docs/testcases/UI/Advanced_Search/1837_4.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+4. Open **Attributes** panel and click **+Add** button.
+5. Click into the **Key** field and select `Dict2` name from dropdown list.
+6. Click into the **Value** field and select `Value2` from dropdown list. Click **Add** button.
+
+**Actions**:
+
+1. Click on **Search** icon at the left menu bar.
+2. Navigate to `Dict2` filter.
+3. Select `Value2` control.
+4. Hover over the found `storage1` in an appeared list
```
Comment (NShaforostov) [inline docs/testcases/UI/Advanced_Search/1837_4.md]: Change the action to the actual current one (_click the 'info' icon_)

File: docs/testcases/UI/Advanced_Search/1837_5.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+4. Open **Attributes** panel and click **+Add** button.
+5. Click into the **Key** field and select `Dict2` name from dropdown list.
+6. Click into the **Value** field and select `Value2` from dropdown list. Click **Add** button.
+
+**Actions**:
+
+1. Click on **Search** icon at the left menu bar.
+2. Navigate to `Dict2` filter.
+3. Select `Value2` control.
+4. Hover over the found `pipeline1` in an appeared list
```
Comment (NShaforostov) [inline docs/testcases/UI/Advanced_Search/1837_5.md]: Change the action to the actual current one (_click the 'info' icon_)

File: docs/testcases/UI/Advanced_Search/1837_6.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+5. Open **Attributes** panel and click **+Add** button.
+6. Click into the **Key** field and select `Dict3` name from dropdown list.
+7. Click into the **Value** field and select `Value3` from dropdown list. Click **Add** button.
+
+**Actions**:
+
+1. Click on **Search** icon at the left menu bar.
+2. Navigate to `Dict3` filter.
+3. Select `Value3` control.
+4. Hover over the found `file1` in an appeared list
```
Comment (NShaforostov) [inline docs/testcases/UI/Advanced_Search/1837_6.md]: Change the action to the actual current one (_click the 'info' icon_)

File: docs/testcases/UI/Advanced_Search/1837_7.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+4. Open **Attributes** panel and click **+Add** button.
+5. Click into the **Key** field and select `Dict1` name from dropdown list.
+6. Click into the **Value** field and select `Value1` from dropdown list. Click **Add** button.
+
+**Actions**:
+
+1. Click on **Search** icon at the left menu bar.
+2. Navigate to `Dict1` filter.
+3. Select `Value1` control.
+4. Hover over the found `tools_group1` in an appeared list
```
Comment (NShaforostov) [inline docs/testcases/UI/Advanced_Search/1837_7.md]: Change the action to the actual current one (_click the 'info' icon_)

File: docs/testcases/UI/Advanced_Search/1837_8.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+4. Open **Attributes** panel and click **+Add** button.
+5. Click into the **Key** field and select `Dict1` name from dropdown list.
+6. Click into the **Value** field and select `Value1` from dropdown list. Click **Add** button.
+
+**Actions**:
+
+1. Click on **Search** icon at the left menu bar.
+2. Navigate to `Dict1` filter.
+3. Select `Value1` control.
+4. Hover over the found `tool1` in an appeared list
```
Comment (NShaforostov) [inline docs/testcases/UI/Advanced_Search/1837_8.md]: Change the action to the actual current one (_click the 'info' icon_)

File: docs/testcases/UI/Advanced_Search/1837_8.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
+# Search for Tools Group by the attributes.
```
Comment (NShaforostov) [inline docs/testcases/UI/Advanced_Search/1837_8.md]: Change the test case header

**Review by NShaforostov — CHANGES_REQUESTED** [review]
Fix marked changes, please.

File: docs/testcases/UI/Advanced_Search/1837_3.md (hunk 1 above)
Comment (maryvictol) [reply docs/testcases/UI/Advanced_Search/1837_3.md]: Done

File: docs/testcases/UI/Advanced_Search/1837_3.md (hunk 2 above)
Comment (maryvictol) [reply docs/testcases/UI/Advanced_Search/1837_3.md]: Done

File: docs/testcases/UI/Advanced_Search/1837_3.md (hunk 3 above)
Comment (maryvictol) [reply docs/testcases/UI/Advanced_Search/1837_3.md]: Done

File: docs/testcases/UI/Advanced_Search/1837_8.md (hunk 9 above)
Comment (maryvictol) [reply docs/testcases/UI/Advanced_Search/1837_8.md]: Done

File: docs/testcases/UI/Advanced_Search/1837_8.md (hunk 8 above)
Comment (maryvictol) [reply docs/testcases/UI/Advanced_Search/1837_8.md]: Done

File: docs/testcases/UI/Advanced_Search/1837_7.md (hunk 7 above)
Comment (maryvictol) [reply docs/testcases/UI/Advanced_Search/1837_7.md]: Done

File: docs/testcases/UI/Advanced_Search/1837_6.md (hunk 6 above)
Comment (maryvictol) [reply docs/testcases/UI/Advanced_Search/1837_6.md]: Done

File: docs/testcases/UI/Advanced_Search/1837_5.md (hunk 5 above)
Comment (maryvictol) [reply docs/testcases/UI/Advanced_Search/1837_5.md]: Done

File: docs/testcases/UI/Advanced_Search/1837_4.md (hunk 4 above)
Comment (maryvictol) [reply docs/testcases/UI/Advanced_Search/1837_4.md]: Done

File: docs/testcases/UI/Advanced_Search/1837_1.md [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+| 5 | Click "Save" button | |
+| 6 | Click on **Search** icon at the left menu bar | Faceted Filter Search page opens. Faceted filters at the left of the page include `Dict1` - `Dict4` filters:<ul><li>Faceted filter controls are sorted by `order` : `Dict1, Dict4, Dict2, Dict3` </li><li> All entries are shown for `Dict1`, 4 entries - for `Dict2`, 3 entries - `Dict3` and 5 entries for `Dict4` </li><li> **Show all** link is displayed for `Dict2` - `Dict4` to expand all dictionary entries </ul></li>|
+| 7 | Click **Show all** link for `Dict2` | <li>List of all entries for `Dict2` is displayed </li><li> **Show all** link is changed to **Hide** link </li>|
+| 8 | Click **Hide** link for `Dict2` | List entries for `Dict2` is collapsed to initial view |
+| 9 | Click arrow near the `Dict1` name | Filter control `Disc1` is collapsed |
```
Comment (NShaforostov) [inline docs/testcases/UI/Advanced_Search/1837_1.md]: Change `Disc1` in **Expected results** to `Dict1`

File: docs/testcases/UI/Advanced_Search/1837_1.md [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+| 10 | Click arrow near the `Dict1` name again | Filter control `Disc1` is expanded to previous view |
```
Comment (NShaforostov) [inline docs/testcases/UI/Advanced_Search/1837_1.md]: Change `Disc1` in **Expected results** to `Dict1`

File: docs/testcases/UI/Advanced_Search/1837_2.md [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+- Tool `tool1`
+
+| Steps | Actions | Expected results |
+| :---: | --- | --- |
+| 1 | Login as admin user from the prerequisites
+| 2 | Open **Preferences** tab in System settings
+| 3 | Click on **Faceted Filter** menu item on the left panel
+| 4 | Add<br>`{`<br><ul>`"dictionary": "Dict5",`<br>   `"defaultDictEntriesToDisplay": "All"`<br></ul>`}`<br>into `faceted.filter.dictionaries` json | |
+| 5 | Click **Save** button | |
+| 6 | Click on **Search** icon at the left menu bar | Advanced Search page opens:<ul><li>List of all search results corresponding search by `*` is shown on Search results page. </li><li> Faceted filters panel: </li><ul><li>Counts of associeted objects are shown near the each entry </li><li> Entries are sorted by count descending inside each filter </li><li> Filter `Dict5` isn't displayed on the Faceted filters panel |
```
Comment (NShaforostov) [inline docs/testcases/UI/Advanced_Search/1837_2.md]: Fix the spelling of the "associated objects"

File: docs/testcases/UI/Advanced_Search/1837_3.md [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+| 2 | Navigate to `Dict1` filter. | The count of the objects associated with `Value1` is more or equal 1 |
+| 3 | Select `Value1` control. | <li>List with the search results that contains at least of `folder1` appears</li><li>**FOLDERS** button is shown count of found entities that is more or equal 1 </li>|
+| 4 | Click the `info` icon for the found `folder1` in an appeared list | Preview panel appears that contains:<ul><li>Folder name that is equal to the name `folder1`</li><li>Attribute `Dict1` with value `Value1`</li><li>`pipeline1` and `storage1` created at Preparations |
+| 5 | `Ctrl +` Click on the found `folder1` name | The `folder1` page opens in the new browser tab |
+| 6 | Click **Table** button to switch page to table view | `Value1` is shown in `Dict1` column for `folder1` |
+| 7 | Click **Clear filters** button | |
+| 8 | Enter `folder1` name into the search string | <li>List with the search results that contains at least of `folder1` appears</li><li>**FOLDERS** button is shown count of found entities that is more or equal 1</li><li>For the `Dict1` filter count of the objects associated with `Value1` control is more or equal 1 </li>|
```
Comment (NShaforostov) [inline docs/testcases/UI/Advanced_Search/1837_3.md]: Seems that described results should be after the step 9

File: docs/testcases/UI/Advanced_Search/1837_4.md [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+
+**Prerequisites**:
+- admin user
+- Perform [1837_3](1837_3.md)
+
+**Preparations:**
+
+1. Login as admin user from the prerequisites.
+2. Open **Library** page.
+3. Open storage (`storage1`) created at Preparations of 1837_3.
```
Comment (NShaforostov) [inline docs/testcases/UI/Advanced_Search/1837_4.md]: Add link to 1837_3 case

File: docs/testcases/UI/Advanced_Search/1837_5.md [hunk 15]
```diff
@@ -0,0 +1,29 @@
...
+
+**Prerequisites**:
+- admin user
+- Perform [1837_3](1837_3.md) case
+
+**Preparations:**
+
+1. Login as admin user from the prerequisites.
+2. Open **Library** page.
+3. Open pipeline (`pipeline1`) created at Preparations of 1837_3.
```
Comment (NShaforostov) [inline docs/testcases/UI/Advanced_Search/1837_5.md]: Add link to 1837_3 case

File: docs/testcases/UI/Advanced_Search/1837_6.md [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+**Prerequisites**:
+- admin user
+- Perform [1837_3](1837_3.md) case
+
+**Preparations:**
+
+1. Login as admin user from the prerequisites.
+2. Open **Library** page.
+3. Open `storage1`
+4. Open file (`file1`) created at Preparations of 1837_3.
```
Comment (NShaforostov) [inline docs/testcases/UI/Advanced_Search/1837_6.md]: Add link to 1837_3 case

File: docs/testcases/UI/Advanced_Search/1838_1.md [hunk 17]
```diff
@@ -0,0 +1,20 @@
+# Advanced Search results: Default Table view for different entities
+
+Test verifies default Advanced Search results Table view for each entity tpes.
```
Comment (NShaforostov) [inline docs/testcases/UI/Advanced_Search/1838_1.md]: Fix the spelling of "entity type"

File: docs/testcases/UI/Advanced_Search/1837_1.md [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+| :---: | --- | --- |
+| 1 | Login as admin user from the prerequisites | |
+| 2 | Open **Preferences** tab in System settings
+| 3 | Click on **Faceted Filter** menu item on the left panel
+| 4 | Set json<br />`{`<br>`"defaultDictEntriesToDisplay": 4,`<br> `"dictionaries": [`<br>`{`<ul>`"dictionary": "Dict1",`<br>`"order": 1,`<br>`"defaultDictEntriesToDisplay": "All"`</ul>`}, {`<ul>`"dictionary": "Dict2",`<br>`"order": 4`</ul>`}, {`<ul>`"dictionary": "Dict3",`<br>    `"defaultDictEntriesToDisplay": 3`</ul>  `}, {`<ul> `"dictionary": "Dict4",`<br>`"order": 2,`<br>`"defaultDictEntriesToDisplay": 5`</ul> ` } ]`<br>`}`<br>into `faceted.filter.dictionaries` field | |
+| 5 | Click "Save" button | |
+| 6 | Click on **Search** icon at the left menu bar | Faceted Filter Search page opens. Faceted filters at the left of the page include `Dict1` - `Dict4` filters:<ul><li>Faceted filter controls are sorted by `order` : `Dict1, Dict4, Dict2, Dict3` </li><li> All entries are shown for `Dict1`, 4 entries - for `Dict2`, 3 entries - `Dict3` and 5 entries for `Dict4` </li><li> **Show all** link is displayed for `Dict2` - `Dict4` to expand all dictionary entries </ul></li>|
```
Comment (NShaforostov) [inline docs/testcases/UI/Advanced_Search/1837_1.md]: Change "Faceted Filter Search page" to "Advanced Search page"

File: docs/testcases/UI/Advanced_Search/1838_2.md [hunk 19]
```diff
@@ -0,0 +1,20 @@
...
+- Dictionaries `Dict1`, `Dict2`, `Dict3`
+
+| Steps | Actions | Expected results |
+| :---: | --- | --- |
+| 1 | Login as admin user from the prerequisites | |
+| 2 | Open **Preferences** tab in System settings | |
+| 3 | Click on **Search** menu item on the left panel | |
+| 4 | Specify list additional columns for preference `search.elastic.index.metadata.fields` as array of strings: `["Dict1","Dict2","Dict3"]` | |
+| 5 | Click "Save" button | |
+| 6 | Click on **Search** icon at the left menu bar | Faceted Filter Search page opens:<ul><li>List of all search results corresponding search by `*` is shown on Search results page.</li><li>Each result is presented by the "row" in the list with next info: ***Name*** and ***Owner*** (if exist)</li> |
```
Comment (NShaforostov) [inline docs/testcases/UI/Advanced_Search/1838_2.md]: Change "Faceted Filter Search page" to "Advanced Search page"

File: docs/testcases/UI/Advanced_Search/1838_3.md [hunk 20]
```diff
@@ -0,0 +1,18 @@
...
+- admin user
+
+| Steps | Actions | Expected results |
+| :---: | --- | --- |
+| 1 | Login as admin user from the prerequisites | |
+| 2 | Click on **Search** icon at the left menu bar | |
+| 3 | Click **Table** button to switch page to table view | <li>All results are presented by the single table</li><li>Table has ***Name, Owner, Description, Changed, Path, Size, Started, Finished*** columns by default </li>|
+| 4 | Logout | |
+| 5 | Login as admin user again | |
+| 6 | Click on **Search** icon at the left menu bar again | Faceted Filter Search page opens:<ul><li>**Table** format is selected</li><li>All results are presented in Table view </li>|
```
Comment (NShaforostov) [inline docs/testcases/UI/Advanced_Search/1838_3.md]: Change "Faceted Filter Search page" to "Advanced Search page"

File: docs/testcases/UI/Advanced_Search/1838_3.md [hunk 21]
```diff
@@ -0,0 +1,18 @@
...
+| 7 | Click **List** button to switch page to list view | |
+| 8 | Repeat steps 3-5 | Faceted Filter Search page opens:<ul><li>**List** format is selected</li><li>Each result is presented by the "row" in the list with next info: ***Name*** and ***Owner*** (if exist) </li>|
```
Comment (NShaforostov) [inline docs/testcases/UI/Advanced_Search/1838_3.md]: Seems that there should be "Repeat steps 4-6".

Change "Faceted Filter Search page" to "Advanced Search page" in **Expected results**.

File: docs/testcases/UI/Advanced_Search/1838_3.md [hunk 22]
```diff
@@ -0,0 +1,18 @@
+# Keeping of the Search results format for the current user.
+
+Test verifies that last selected Search results format are saved for the current user and used for the next time they will leverage the Advanced Search.
```
Comment (NShaforostov) [inline docs/testcases/UI/Advanced_Search/1838_3.md]: Change description to:
"Test verifies that the selected format of the Advanced Search results (list/table) is saved for the current user until the next Search usage."

**Review by NShaforostov — CHANGES_REQUESTED** [review]
Fix changes, please.

Also common note to cases 1837_3, 1837_4, 1837_5, 1837_6, 1837_7, 1837_8:
add, please, an additional step after the preview opening, on which the preview is being closed (previously, it was not needed due to other preview type).

**Review by NShaforostov — APPROVED** [review]

**Pull request merged**